### PR TITLE
refactor: Consolidate Android and JVM calling code in logic module

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues
A follow up PR of #1412 to consolidate Android and JVM code
### Solutions
I removed the duplicated code and added a new class called `CurrentPlatform` to use it to initialise AVS. 

That's because we init AVS in different ways for Android and JVM and the `Calling` interface is placed inside `commonJvmAndroid` directory 

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/util/CurrentPlatform.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+package com.wire.kalium.logic.util
+
+actual class CurrentPlatform(
+    val name: PlatformType = PlatformType.ANDROID
```
Comment (alexandreferris) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/util/CurrentPlatform.kt:23]: maybe we can change `name` to `type` to better express what we have there

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/MediaManagerServiceImpl.kt [hunk 2]
```diff
@@ -18,19 +18,19 @@
...
 import com.waz.media.manager.MediaManagerListener
 import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.logic.util.PlatformContext
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 
 actual class MediaManagerServiceImpl(
-    appContext: Context
+    appContext: PlatformContext
```
Comment (alexandreferris) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/MediaManagerServiceImpl.kt:30]: maybe we can rename this to `platformContext`

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/FlowManagerServiceImpl.kt [hunk 3]
```diff
@@ -18,23 +18,24 @@
...
+import com.wire.kalium.logic.util.PlatformContext
 import com.wire.kalium.logic.util.PlatformView
 import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 actual class FlowManagerServiceImpl(
-    appContext: Context,
-    private val dispatcher: CoroutineDispatcher
+    appContext: PlatformContext,
```
Comment (alexandreferris) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/feature/call/FlowManagerServiceImpl.kt:32]: maybe we can rename this to `platformContext`

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/util/PlatformContext.kt [hunk 4]
```diff
@@ -16,17 +16,10 @@
...
-    override fun onVideoReceiveStateChanged(conversationId: String, userId: String, clientId: String, state: Int, arg: Pointer?) {
-        callingLogger.i(
-            "[onVideoReceiveStateChanged] - conversationId: $conversationId | userId: $userId clientId: $clientId" +
-                    " | state: $state"
-        )
-    }
-}
+actual class PlatformContext(
+    val context: Context
+)
```
Comment (alexandreferris) [inline logic/src/androidMain/kotlin/com/wire/kalium/logic/util/PlatformContext.kt:25]: Strange enough this one looks like it removed `OnParticipantsVideoStateChanged` and renamed/added `PlatformContext` 🤔

File: logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt [hunk 5]
```diff
@@ -153,7 +153,7 @@ class CallManagerImpl internal constructor(
                     messageSender,
                     scope,
                     callMapper
-                ).keepingStrongReference(),
+                ),
```
Comment (alexandreferris) [inline logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:156]: Why are we not `keepingStrongReference` of `OnSendOTR` handler? 🤔

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/util/CurrentPlatform.kt [hunk 6]
```diff
@@ -14,18 +14,15 @@
...
 
-object OnMissedCall : MissedCallHandler {
-    override fun onMissedCall(conversationId: String, messageTime: Uint32_t, userId: String, isVideoCall: Boolean, arg: Pointer?) {
-        // NOTHING TO DO | This callback is not triggered by AVS
-        callingLogger.i("[onMissedCall] - conversationId: $conversationId | userId: $userId")
-    }
+enum class PlatformType(val value: String) {
+    ANDROID("Android"),
+    JVM("Jvm"),
+    DARWIN("Darwin")
```
Comment (alexandreferris) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/util/CurrentPlatform.kt:27]: Strange enough this one looks like it removed `OnMissedCall` and renamed/added `CurrentPlatform` 🤔

**Review by alexandreferris — COMMENTED** [review]
Great work moving everything @ohassine 🙌🏻 

I just have a couple of comments (mostly trying to understand), and if possible it would also be good to have a second look from @typfel as if I remember he did some stuff to make calling work on CLI? Or maybe I'm just crazy

File: logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt (hunk 5 above)
Comment (vitorhugods) [inline logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:156]: Strong reference not needed anymore?

**Review by vitorhugods — COMMENTED** [review]
Looks good overall, just curious about the removal of `keepingStrongReference` for the self handler

File: logic/src/androidMain/kotlin/com/wire/kalium/logic/util/PlatformContext.kt (hunk 4 above)
Comment (typfel) [reply logic/src/androidMain/kotlin/com/wire/kalium/logic/util/PlatformContext.kt:25]: I think it's just the GitHub diff that's confused, when look at the branch the `OnParticipantsVideoStateChanged` and `OnMissedCall` is still there.

File: logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt (hunk 5 above)
Comment (ohassine) [reply logic/src/commonJvmAndroid/kotlin/com/wire/kalium/logic/feature/call/CallManagerImpl.kt:156]: Oh! that should not be removed. Good catch 👌

**Review by alexandreferris — APPROVED** [review]
👌🏻

**Pull request merged**